### DISCUSSION TITLE
Plans to Bring C++ Back?

### DISCUSSION THREAD

**Discussion by cozkul** [daveight/ankicode#38] [General]
Hi, are there any plans to bring C++ back? Are there any roadblocks? I tried using older v1.2 and I did get some linker errors, was this the main issue that caused C++ being dropped from v1.3?

**Comment by daveight** [comment]
Hey, thanks for reaching out.
Indeed there were problems related to bundling of C++ compiler runtime in the app, especially for MacOS. And that is the reason that I temporarily removed C++ support from the app.

However I would like to bring C++ back and I will take a closer look on this problem. I will keep you posted.

**Comment by cozkul** [comment]
Hi, I think I managed to get C++ working again: I edited `cpp_test_runner.py` and replaced `/testing_libs/vendor/libs/cpp` with gcc-14 and dropped clang. gcc-14 also works with `-std=c++20`. Would you like me to create an issue and submit a PR against it?

Also I couldn't figure out how `/testing_libs/vendor/` is bundled into release. How do you bundle the compilers? (I copy pasted `/testing_libs/vendor/` manually after building with `./run`)

**Reply by daveight** [reply]
Nice! That would be great if you could create the PR.

>gcc-14 
Is there a download link?

>Also I couldn't figure out how /testing_libs/vendor/ is bundled into release. How do you bundle the compilers? (I copy pasted /testing_libs/vendor/ manually after building with ./run)

to bundle the app I used the command:
`./ninja bundle`

But if you will create the PR, I could bundle it and release new version.

Thank you.

**Reply by Tinagon** [reply]
Any progress on this?

**Reply by daveight** [reply]
Unfortunately no progress so far, If you want to help, let me know.

**Reply by cozkul** [reply]
Hi, sorry for the extended delay on this. I had personal responsibilities that kept me very busy. I created a pull request to change the build script. @daveight You'd still need to get binaries elsewhere, I added details to my pull req.

**Reply by daveight** [reply]
Hey @cozkul  @Tinagon ! I've published a pre-release v1.4.0 (https://github.com/daveight/ankicode/releases/tag/ankicode-1.4.0-pre)
I would be appreciated if you can check that C++ integration works as expected (for me it works fine)

Thanks

**Reply by Tinagon** [reply]
> Hey @cozkul @Tinagon ! I've published a pre-release v1.4.0 (https://github.com/daveight/ankicode/releases/tag/ankicode-1.4.0-pre) I would be appreciated if you can check that C++ integration works as expected (for me it works fine)
> 
> Thanks

Thanks a lot for your time and effort in bringing C++ integration back! I really appreciate it. I'll check it out soon!

**Reply by cozkul** [reply]
@daveight it works for me thank you!!
